src/main/java/org/broadinstitute/hellbender/tools/ConvertHadoopBamFragmentToBam.java

```diff
@@ -0,0 +1,88 @@
+package org.broadinstitute.hellbender.tools;
+
+import htsjdk.samtools.*;
+import htsjdk.samtools.util.BinaryCodec;
+import htsjdk.samtools.util.BlockCompressedOutputStream;
+import htsjdk.samtools.util.BlockCompressedStreamConstants;
+import htsjdk.samtools.util.RuntimeIOException;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgram;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.programgroups.TestSparkProgramGroup;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.io.IOUtils;
+
+import java.io.*;
+
+
+/**
+ * Spark troubleshooting utility that converts a Hadoop bam fragment (eg., a part0000, part0001, etc. file
+ * produced by ReadsSparkSink) into a readable bam file by adding a header and a BGZF terminator.
+ */
+@CommandLineProgramProperties(
+        summary = "Convert a hadoop bam fragment into a readable bam",
+        oneLineSummary = "Convert a hadoop bam fragment into a readable bam",
+        programGroup = TestSparkProgramGroup.class
+)
+public final class ConvertHadoopBamFragmentToBam extends CommandLineProgram {
+
+    @Argument(shortName = "hadoopFragment", fullName = "hadoopFragment", doc = "Hadoop BAM fragment to be converted", optional = false)
+    private File hadoopFragment;
+
+    @Argument(shortName = "bamWithHeader", fullName = "bamWithHeader", doc = "Well-formed BAM whose header to use for the converted fragment", optional = false)
+    private File bamWithHeader;
+
+    @Argument(shortName = "O", fullName = "outputBam", doc = "Location to write the converted BAM Fragment", optional = false)
+    private File outputBam;
+
+    @Override
+    protected Object doWork(){
+        SAMFileHeader header = null;
+        try ( final SamReader headerReader = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT).open(bamWithHeader) ) {
+            header = headerReader.getFileHeader();
+        }
+        catch ( IOException e ) {
+            throw new UserException("Error reading header from " + bamWithHeader.getAbsolutePath(), e);
+        }
+
+        try ( FileOutputStream outStream = new FileOutputStream(outputBam) ) {
+            writeHeader(outStream, header);
+            final byte[] fragmentBytes = IOUtils.readFileIntoByteArray(hadoopFragment);
```

Comment:
Well, this utility was written specifically to identify a bug in `ReadsSparkSink#mergeBam`, which is why it doesn't call into it (and for the same reason, I think it should continue to be independent). I'll modify it so that it doesn't slurp the entire fragment into memory at once, though, to allow it to work with larger fragments.